pydantic/main.py

```diff
@@ -1,3 +1,4 @@
+import inspect
```

Comment:
Please more this to `if TYPE_CHECKING` and use `from inspect import Signature`.

Inspect is relatively slow to import, though pydantic makes an effort to only import it when used.

see #1127